File: src/api/routes/price_book.py
"""价格本 API 路由"""
from fastapi import APIRouter, HTTPException, Query
from typing import List, Optional
from src.services.price_book_service import PriceBookService

router = APIRouter(prefix="/api/price-book", tags=["price-book"])
service = PriceBookService()


@router.get("/entries")
async def get_all_entries():
    entries = await service.get_all()
    return entries


@router.get("/entries/{entry_id}")
async def get_entry(entry_id: str):
    entry = await service.get_by_id(entry_id)
    if not entry:
        raise HTTPException(status_code=404, detail="价格本条目未找到")
    return entry


@router.get("/match")
async def match_keyword(keyword: str = Query(...)):
    """通过关键词匹配价格本条目"""
    entry = await service.get_by_keyword(keyword)
    return entry


@router.post("/entries")
async def create_entry(data: dict):
    entry = await service.create(data)
    return {"message": "创建成功", "data": entry}


@router.put("/entries/{entry_id}")
async def update_entry(entry_id: str, data: dict):
    entry = await service.update(entry_id, data)
    if not entry:
        raise HTTPException(status_code=404, detail="价格本条目未找到")
    return {"message": "更新成功", "data": entry}


@router.delete("/entries/{entry_id}")
async def delete_entry(entry_id: str):
    success = await service.delete(entry_id)
    if not success:
        raise HTTPException(status_code=404, detail="价格本条目未找到")
    return {"message": "删除成功"}


@router.put("/batch-update")
async def batch_update_entries(data: dict):
    """批量更新价格本条目"""
    ids = data.get("ids", [])
    if not ids:
        raise HTTPException(status_code=400, detail="请提供要更新的条目ID")
    count = await service.batch_update(ids, data)
    return {"message": f"已更新 {count} 条", "count": count}


@router.post("/evaluate")
async def evaluate_item(data: dict):
    """评估单个商品"""
    keyword = data.get("keyword", "")
    price = data.get("price", 0)
    result = await service.evaluate_item(keyword, price)
    return result


@router.post("/evaluate-batch")
async def evaluate_items_batch(data: dict):
    """批量评估商品"""
    items = data.get("items", [])
    results = await service.evaluate_items_batch(items)
    return results


@router.post("/auto-update-market-prices")
async def auto_update_market_prices():
    """自动更新行情价"""
    await service.auto_update_market_prices()
    return {"message": "行情价更新完成"}

File: web-ui/src/api/priceBook.ts
import type { PriceBookEntry } from '@/types/priceBook'
import { http } from './http'

/** 获取所有价格本条目 */
export async function getAllEntries(): Promise<PriceBookEntry[]> {
  return await http('/api/price-book/entries')
}

/** 获取单个价格本条目 */
export async function getEntry(entryId: string): Promise<PriceBookEntry> {
  return await http(`/api/price-book/entries/${entryId}`)
}

/** 关键词匹配价格本条目 */
export async function matchKeyword(keyword: string): Promise<PriceBookEntry | null> {
  return await http('/api/price-book/match', { params: { keyword } })
}

/** 创建价格本条目 */
export async function createEntry(data: Partial<PriceBookEntry>): Promise<{ message: string; data: PriceBookEntry }> {
  return await http('/api/price-book/entries', {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify(data),
  })
}

/** 更新价格本条目 */
export async function updateEntry(entryId: string, data: Partial<PriceBookEntry>): Promise<{ message: string; data: PriceBookEntry }> {
  return await http(`/api/price-book/entries/${entryId}`, {
    method: 'PUT',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify(data),
  })
}

/** 删除价格本条目 */
export async function deleteEntry(entryId: string): Promise<{ message: string }> {
  return await http(`/api/price-book/entries/${entryId}`, { method: 'DELETE' })
}

/** 批量更新价格本条目 */
export async function batchUpdate(ids: string[], data: Record<string, unknown>): Promise<{ message: string; count: number }> {
  return await http('/api/price-book/batch-update', {
    method: 'PUT',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify({ ids, ...data }),
  })
}

/** 单品评估 */
export async function evaluate(keyword: string, price: number): Promise<Record<string, unknown>> {
  return await http('/api/price-book/evaluate', {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify({ keyword, price }),
  })
}

/** 批量评估 */
export async function evaluateBatch(items: { keyword: string; price: number }[]): Promise<Record<string, unknown>[]> {
  return await http('/api/price-book/evaluate-batch', {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify({ items }),
  })
}

/** 自动更新行情价 */
export async function autoUpdateMarketPrices(): Promise<{ message: string }> {
  return await http('/api/price-book/auto-update-market-prices', {
    method: 'POST',
  })
}
